src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -1538,15 +1577,23 @@ int decryptBlocks(GCM op, byte[] in, int inOfs, int inLen,
             if (bLen > 0) {
                 buffer = ibuffer.toByteArray();
 
-                if (bLen >= blockSize) {
-                    len += op.update(buffer, 0, bLen, out, outOfs);
-                    outOfs += len; // noop for ghash
+                if (bLen >= PARALLEL_LEN) {
+                    len = GaloisCounterMode.implGCMCrypt(buffer, 0, bLen,
+                        buffer, 0, out, outOfs, gctr, ghash);
+                    outOfs += len;
                     // Use len as it becomes the ibuffer offset, if
                     // needed, in the next op
                 }
 
-                // merge the remaining ibuffer with the 'in'
                 int bufRemainder = bLen - len;
+                if (bufRemainder >= blockSize) {
+                    int r = op.update(buffer, len, bufRemainder, out, outOfs);
```

Comment:
nit: it seems that this method uses "resultLen" to store the return value of each update() call. For consistency and better readability, perhaps just one and not both, i.e. reusing "resultLen" or a local variable when needed such as "r".